### ISSUE TITLE
Parser incorrectly identifies let-expr when missing variable declaration

### ISSUE THREAD

**Issue by gayalkuruppu** [ballerina-platform/ballerina-lang#34901]
**Description:**

Consider the following code snippet,
```
public function foo(int a, float b) {
    let float x1 = b + b, int x2 = a i
}
```
parser identifies the let expression as,
```
let float x1 = b + b, int x2 = a  MISSING[,]i
 MISSING[] MISSING[=] MISSING[] MISSING[in] MISSING[]
 ```

**Affected Versions:**
SL 2201.0.0

**Related Issues**
#34746

**Comment by lochana-chathura** [comment]
With the variable assignment, it recovers correctly as,
```
public function foo(int a, float b) {
    var a = let float x1 = b + b, int x2 = a  MISSING[+]i
 MISSING[in] MISSING[] MISSING[;]}
```
and gives the required `in` keyword suggestion. 

We need to give a consistence recovery for without variable assignment case as well.

**Comment by lochana-chathura** [comment]
Fixed in #35066
